res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+    components.Button.call(this, options);
+};
+RDJ2.LoopInButton.prototype = new components.Button({
+    outKey: "loop_start_position",
+    outValueScale: function(value) { return (value >= 0) * this.max; },
+    unshift: function() {
+        this.inKey = "loop_in";
+        this.input = components.Button.prototype.input;
+    },
+    shift: function() {
+        //pressing when shifted will delete loop start marker
+        this.inKey = "loop_start_position";
+        this.input = function(channel, control, value, status) {
+            if (this.isPress(channel, control, value, status)) {
+                this.inSetValue(RDJ2.MIXXX_LOOP_POSITION_UNDEFINED);
+            }
+        };
+    },
+});
+
+RDJ2.LoopOutButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.LoopOutButton.prototype = new components.Button({
+    outKey: "loop_end_position",
+    outValueScale: function(value) { return (value >= 0) * this.max; },
+    unshift: function() {
+        this.inKey = "loop_out";
+        this.input = components.Button.prototype.input;
+    },
+    shift: function() {
+        //pressing when shifted will delete loop end marker
+        this.inKey = "loop_end_position";
+        this.input = function(channel, control, value, status) {
+            if (this.isPress(channel, control, value, status)) {
+                this.inSetValue(RDJ2.MIXXX_LOOP_POSITION_UNDEFINED);
+            }
+        };
+    },
+});
+
+RDJ2.AutoLoopButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.AutoLoopButton.prototype = new components.Button({
+    outKey: "beatloop_activate",
+    unshift: function() {
+        this.inKey = "beatloop_activate";
+    },
+    shift: function() {
+        this.inKey = "beatlooproll_activate";
+    },
+});
+
+RDJ2.LoopActiveButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.LoopActiveButton.prototype = new components.Button({
+    outKey: "loop_enabled",
+    unshift: function() {
+        this.inKey = "reloop_toggle";
+    },
+    shift: function() {
+        this.inKey = "reloop_andstop";
+    },
+});
+
+
+////////////////////////////////////////////////////////////////////////
+// Knobs                                                              //
+////////////////////////////////////////////////////////////////////////
+
+RDJ2.MIDI_KNOB_INC = 0x41;
+RDJ2.MIDI_KNOB_DEC = 0x3F;
+RDJ2.MIDI_KNOB_DELTA_BIAS = 0x40; // center value of relative movements
+//RDJ2.MIDI_KNOB_STEPS = 20;  // 20 is full knob's rotation (360deg)
+RDJ2.MIDI_KNOB_STEPS = 16;    // 16 is more like volume knobs
+
+//currently RDJ2.getKnobDeltaOld is not used
+RDJ2.getKnobDeltaOld = function(midiValue) {
+    switch (midiValue) {
+    case RDJ2.MIDI_KNOB_INC:
+        return 1;
+    case RDJ2.MIDI_KNOB_DEC:
+        return -1;
+    default:
+        RDJ2.logError("Unexpected MIDI knob value: " + midiValue);
+        return 0;
+    }
+};
```

Comment:
I'd suggest to remove this method if its not used.

Reply:
Done in the latest commit.